src/main/java/org/openpnp/machine/reference/ReferencePnpJobProcessor.java

```diff
@@ -426,8 +427,9 @@ protected void doPlan() throws Exception {
         fireTextStatus("Planning placements.");
 
         // Get the list of unfinished placements and sort them by part height.
+
         List<JobPlacement> jobPlacements = getPendingJobPlacements().stream()
-                .sorted(Comparator.comparing(JobPlacement::getPartHeight))
+                .sorted(Comparator.comparing(JobPlacement::getPartId))
```

Comment:
This is a major change to how the order of placements works and I think it needs to be discussed before adding this. The reason we use height is so that the machine doesn't run into tall parts after they are placed

Reply:
oh, i didnt see any problems with. already done a lot of board with it. order are more predictable and can be set by table sorting. very good feature.

with PartHeight order, absolutely not predicable and not user friendly order.

Reply:
it's a good idea, to take the part height into account. Perhaps this value should be listed, so the user can sort the list by the parts height. Or at least he has it in focus, before starting the job. 

Another idea could be, that the user gets warned when starting a job with "mixed" heights. A configurable threshold should help to give it some tolerance (e.g. 0.2mm). 

Third Idea: I would like to group e.g. all resistors, then all ceramic caps, then diodes, then ICs, ... Similar to a manual pick and place. I would like to select multiple lines and define a group (floating point number). Then one can assign that number (normally integer) as he wants, and then sort the whole list by the grouping number. Special parts can then be handled by using further digits of that floating point number. 

@aneox: what is you typical column you are sorting the table?

Reply:
@aneox @KarlZeilhofer One thing I want to clarify: The note in the PR said "Some changes in job order. Now user can change job order by sorting table." but as far as I can tell this is not the case. It was just sorting by placement ID. The old behavior is by height, and the new behavior was by placement ID. I stuck with the old behavior.

Reply:
@KarlZeilhofer 
for me, best Part column. Job goes from one feeder to another and never return back to previous feeders. easy to control proccess.

@vonnieda 
sorry, i didnt test another columns. I used only part column and this order works well.
Maybe you misunderstand "sorting by placement ID", job should go thought placement ID column. However you can sort table by any column. I cant test now, please play with table column sorting with my code, it should work, as i remember.

Reply:
Okay, I see now that it uses partId, rather than placementId. That is better, but I still can't allow this change in until users can change it themselves. This completely changes the order that jobs progress in, and I can't make that decision for people who have been using OpenPnP for a long time and who depend on it being ordered by height.

My goal is to eventually allow users to set order themselves, in which case the point is moot.

If you feel strongly about this, you could add a setting to the ReferencePnpJobProcessor wizard so that users can select how the ordering happens.

Reply:
@vonnieda 
As KarlZeilhofer said, good idea to add partHeight column and use it to sort by default. Then user can choose themself.

Reply:
Hi @aneox and @KarlZeilhofer: I've tested the code twice and it does not follow the table sorting, it only follows the partId sorting. I don't see how it could. The only change you made was to change the sort in the JobProcessor from partHeight to partId. 

The table is a Swing view, and the sorting is only reflected in the view. It does not reorder anything in the model. The only way to get the sort order from the table is to query it directly, and the JobProcessor can't do that. 

If I've missed something, please let me know. Perhaps there was a change you didn't submit?

Here is a video showing the code from this PR with three different sorts. As you can see, the only sort applied is the partId.

https://www.youtube.com/watch?v=vOBkw_ljq44&feature=youtu.be

Reply:
@vonnieda 
Yop, thank you for testing and i am sorry for confusion. I was using only part column sorting and it seemed like it works by table sorting.